src/coprocessor/endpoint.rs

```diff
@@ -63,14 +64,14 @@ const OUTDATED_ERROR_MSG: &'static str = "request outdated.";
 
 const ENDPOINT_IS_BUSY: &'static str = "endpoint is busy";
 
-pub struct Host<R: CopSender + 'static> {
+pub struct Host {
```

Comment:
use another PR for refactoring coprocessor.

Reply:
Btw, you shouldn't need to let coprocessor know the PD worker directly. You can still pass a sender so you can test the coprocessor independently and easily.